### ISSUE TITLE
DICUSSION : javascript dependency management

### ISSUE THREAD

**Issue by biconou** [Libresonic/libresonic#360]
I'd  like to introduce a solution for JavaScript lib dependency management. 
I think a choice should be made between 

- NPM
- bower

Any suggestion ?

**Comment by muff1nman** [comment]
This might be a good idea, but it is definitely a departure from the existing pattern of packaging json files directly into libresonic. So I think we should even backup a step and start with the question, do we need a Javascript dependency management tool?

**Comment by fxthomas** [comment]
_The_ big benefit I see for adopting a JS dependency management tool is for when we want to upgrade our libraries. Bear in mind that I only tried both of those a few times, but I would favor Bower over NPM with our current setup :

* Bower is marketed as front-end-oriented (and therefore framework agnostic). I believe NPM will work too, but I've run into a few issues with CommonJS modules outside of a full Node and/or Browserify environment.
* Flat directory structure similar to what we currently have in `libresonic-main/src/main/webapp/script`, so nothing changes much.
* NPM's fancy multi-version dependencies resolution isn't very useful in our case because we have very few modules.

*TL;DR:* Bower is simpler, NPM has more features for complex projects.

_([Yarn](https://yarnpkg.com/en/) also seems like an interesting choice I have not explored yet.)_

**Comment by fxthomas** [comment]
Also, I believe we should still check JS modules in the Git repository like we currently do, it is so much easier for people who just want to contribute small features without touching JS code.
